pkg/server/server.go

```
@@ -155,6 +167,88 @@ func (s *Server) getAgentConfig(c echo.Context) error {
+		slog.Error("Failed to read agent YAML file", "path", agentPath, "error", err)
+		return c.JSON(http.StatusInternalServerError, map[string]string{"error": "failed to read agent file"})
+	}
+
+	return c.String(http.StatusOK, string(yamlContent))
+}
+
+func (s *Server) editAgentConfigYAML(c echo.Context) error {
+	agentID := c.Param("id")
+	agentPath, err := s.secureAgentPath(agentID)
+	if err != nil {
+		slog.Error("Invalid agent ID", "agentID", agentID, "error", err)
+		return c.JSON(http.StatusBadRequest, map[string]string{"error": "invalid agent id"})
+	}
+
+	// Check if the file exists
+	if _, err := os.Stat(agentPath); os.IsNotExist(err) {
+		slog.Error("Agent file does not exist", "path", agentPath)
+		return c.JSON(http.StatusNotFound, map[string]string{"error": "agent file not found"})
+	}
+
+	// Read the YAML content from request body
+	body, err := io.ReadAll(c.Request().Body)
+	if err != nil {
+		slog.Error("Failed to read request body", "error", err)
+		return c.JSON(http.StatusBadRequest, map[string]string{"error": "failed to read request body"})
+	}
+	yamlContent := string(body)
+
+	// Validate the YAML content by attempting to parse it
+	// Use ReferenceDirs to allow YAML files to reference other files relative to the agent directory
+	agentDir := filepath.Dir(agentPath)
+	var tmpConfig latest.Config
+	if err := yaml.UnmarshalWithOptions([]byte(yamlContent), &tmpConfig, yaml.Strict(), yaml.ReferenceDirs(agentDir)); err != nil {
+		slog.Error("Invalid YAML content", "error", err)
+		return c.JSON(http.StatusBadRequest, map[string]string{"error": "invalid YAML content: " + err.Error()})
+	}
+
+	// Reload the agent to update the in-memory configuration
+	t, err := teamloader.Load(c.Request().Context(), agentPath, s.runConfig)
+	if err != nil {
+		slog.Error("Failed to reload agent after YAML edit", "path", agentPath, "error", err)
+		return c.JSON(http.StatusInternalServerError, map[string]string{"error": "failed to reload agent configuration"})
```

do we want to surface the exact error to the user of the api so they can fix the file without checking the logs?